### PR TITLE
Changelog for Azure.Core release

### PR DESCRIPTION
**Pull request by christothes**
# Contributing to the Azure SDK

Please see our [CONTRIBUTING.md](https://github.com/Azure/azure-sdk-for-net/blob/main/CONTRIBUTING.md) if you are not familiar with contributing to this repository or have questions.

For specific information about pull request etiquette and best practices, see [this section](https://github.com/Azure/azure-sdk-for-net/blob/main/CONTRIBUTING.md#pull-request-etiquette-and-best-practices).

### CODE REVIEW

File: sdk/core/Azure.Core/CHANGELOG.md [hunk 1]
```diff
@@ -1,12 +1,13 @@
...
 
-## 1.23.0-beta.1 (Unreleased)
+## 1.23.0 (2022-04-05)
 
 ### Features Added
 
 - Added `AddClassifier` methods to `RequestContext`. These methods allow callers to change the response classification behavior for a given method invocation.
 - Added type `RequestOptions` to the `Azure` namespace and made `RequestContext` a subclass of `RequestOptions`.  This enables `RequestOptions` to be exposed in methods that take `CancellationToken` without causing confusion regarding which cancellation token will take effect.
 - Added the `SetUserAgentString` extension method to `HttpMessage` accepting a `UserAgentValue` and an optional application Id string. This allows assembly specific user agent header information to be set with proper formatting on a per-message basis.
+- Added an extension method to `BinaryData` called `ToObjectFromJson` which converts the json value represented by `BinaryData` to an object of a specific type.
```
Comment (annelo-msft) [inline sdk/core/Azure.Core/CHANGELOG.md:10]: Do we want to add anything about `WaitUntil` or the other management plane chagnes?

File: sdk/core/Azure.Core/CHANGELOG.md [hunk 2]
```diff
@@ -1,12 +1,13 @@
 # Release History
 
-## 1.23.0-beta.1 (Unreleased)
+## 1.23.0 (2022-04-05)
 
 ### Features Added
 
 - Added `AddClassifier` methods to `RequestContext`. These methods allow callers to change the response classification behavior for a given method invocation.
 - Added type `RequestOptions` to the `Azure` namespace and made `RequestContext` a subclass of `RequestOptions`.  This enables `RequestOptions` to be exposed in methods that take `CancellationToken` without causing confusion regarding which cancellation token will take effect.
```
Comment (annelo-msft) [inline sdk/core/Azure.Core/CHANGELOG.md:8]: This is wrong -- will fix.

File: sdk/core/Azure.Core/CHANGELOG.md [hunk 3]
```diff
@@ -1,12 +1,13 @@
...
 - Added the `SetUserAgentString` extension method to `HttpMessage` accepting a `UserAgentValue` and an optional application Id string. This allows assembly specific user agent header information to be set with proper formatting on a per-message basis.
```
Comment (annelo-msft) [inline sdk/core/Azure.Core/CHANGELOG.md:9]: This has been renamed

File: sdk/core/Azure.Core/CHANGELOG.md (hunk 2 above)
Comment (annelo-msft) [inline sdk/core/Azure.Core/CHANGELOG.md:8]:
```suggestion
```

File: sdk/core/Azure.Core/CHANGELOG.md [hunk 4]
```diff
@@ -1,12 +1,13 @@
 # Release History
 
-## 1.23.0-beta.1 (Unreleased)
+## 1.23.0 (2022-04-05)
 
 ### Features Added
 
 - Added `AddClassifier` methods to `RequestContext`. These methods allow callers to change the response classification behavior for a given method invocation.
```
Comment (annelo-msft) [inline sdk/core/Azure.Core/CHANGELOG.md:7]:
```suggestion
- Added `AddClassifier` methods to `RequestContext`. These methods allow callers to change the response classification behavior for a given method invocation.
- Added a new `StatusCodeClassifier` type that will be used as the default `ResponseClassifier` for some libraries.
```

File: sdk/core/Azure.Core/CHANGELOG.md (hunk 1 above)
Comment (christothes) [reply sdk/core/Azure.Core/CHANGELOG.md:10]: cc @m-nash

Comment (m-nash) [reply sdk/core/Azure.Core/CHANGELOG.md:9]: I think adding anything about `WaitUntil` here would be out of context, the [samples around LRO](https://github.com/Azure/azure-sdk-for-net/blob/main/sdk/core/Azure.Core/samples/LongRunningOperations.md) should have these updates though for sure.

Comment (annelo-msft) [reply sdk/core/Azure.Core/CHANGELOG.md:10]: https://apiview.dev/Assemblies/Review/77bdbcef49a3465fa96fac3fb745bf57/3547e2b6c9504989bbc669abdc5001ad?diffRevisionId=bf47f1ad7e134e81a1fefe7570186f0f&doc=False&diffOnly=True

Comment (m-nash) [reply sdk/core/Azure.Core/CHANGELOG.md:10]: Ah yeah we can say we added more common public cloud references to `AzureLocation`

File: sdk/core/Azure.Core/CHANGELOG.md [hunk 5]
```diff
@@ -1,12 +1,13 @@
 # Release History
 
-## 1.23.0-beta.1 (Unreleased)
+## 1.23.0 (2022-04-05)
```
Comment (JoshLove-msft) [inline sdk/core/Azure.Core/CHANGELOG.md:3]:
```suggestion
## 1.23.0 (2022-03-21)
```

**Review by JoshLove-msft — APPROVED** [review]

**Review by m-nash — APPROVED** [review]

**Review by annelo-msft — APPROVED** [review]

**Pull request merged**